cmd/disk-cache.go

```diff
@@ -228,18 +229,27 @@ func (c *cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 				dcache.Delete(ctx, bucket, object)
 			}
 		}
+		if cacheErr == nil {
```

Comment:
can this if condition be moved after line 223 and remove the cacheReader.Close() on line 226 ? Otherwise it would lead to attempting to close a reader twice if ObjectNotFound

Reply:
Currently `cacheReader.Close` is idempotent, but I can move it upwards.

Reply:
Done